src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -0,0 +1,746 @@
+    private static final AtomicBoolean loadJobTotalRan = new AtomicBoolean();
+    private static final int TICK_INTERVAL = 15000;
+
+    @Initializer(before=InitMilestone.JOB_LOADED, fatal=false)
+    public static void loadJobTotal() {
+        if (!loadJobTotalRan.compareAndSet(false, true)) {
+            return; // TODO why does Jenkins run the initializer many times?!
+        }
+        scan(new File(Jenkins.getActiveInstance().getRootDir(), "jobs"), 0);
+        // TODO reset count after reload config from disk (otherwise goes up to 200% etc.)
+    }
+    private static void scan(File d, int depth) {
+        File[] projects = d.listFiles();
+        if (projects == null) {
+            return;
+        }
+        for (File project : projects) {
+            if (!new File(project, "config.xml").isFile()) {
+                continue;
+            }
+            if (depth > 0) {
+                jobTotal.incrementAndGet();
+            }
+            File jobs = new File(project, "jobs"); // cf. getJobsDir
+            if (jobs.isDirectory()) {
+                scan(jobs, depth + 1);
+            }
+        }
+    }
+
+    /** Child items, keyed by {@link Item#getName}. */
+    protected transient Map<String,I> items = new CopyOnWriteMap.Tree<String,I>(CaseInsensitiveComparator.INSTANCE);
+
+    private DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor> properties;
+
+    /**
+     * {@link View}s.
+     */
+    private /*almost final*/ CopyOnWriteArrayList<View> views;
+
+    /**
+     * Currently active Views tab bar.
+     */
+    private volatile ViewsTabBar viewsTabBar;
+
+    /**
+     * Name of the primary view.
+     */
+    private volatile String primaryView;
+
+    private transient /*almost final*/ ViewGroupMixIn viewGroupMixIn;
+
+    private DescribableList<FolderHealthMetric,FolderHealthMetricDescriptor> healthMetrics;
+
+    private FolderIcon icon;
+
+    /** Subclasses should also call {@link #init}. */
+    protected AbstractFolder(ItemGroup parent, String name) {
+        super(parent, name);
+    }
+
+    protected void init() {
+        if (properties == null) {
+            properties = new DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor>(this);
+        }
+        for (AbstractFolderProperty p : properties) {
+            p.setOwner(this);
+        }
+        if (icon == null) {
+            icon = new StockFolderIcon();
+        }
+        if (views == null) {
+            views = new CopyOnWriteArrayList<View>();
+        }
+        if (views.isEmpty()) {
+            try {
+                initViews(views);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Failed to set up the initial view", e);
+            }
+        }
+        if (viewsTabBar == null) {
+            viewsTabBar = new DefaultViewsTabBar();
+        }
+        if (primaryView == null) {
+            primaryView = views.get(0).getViewName();
```

Comment:
I see earlier that `initViews` can fail and you fall through. Don't you need to check again if the `views` list `isEmpty`?